### ISSUE TITLE
Incompatibility between Bugsnag Gradle Plugin 7.1.0 and DexGuard 1.1.14/1.1.15

### ISSUE THREAD

**Issue by guilherme-pereira-ifood** [bugsnag/bugsnag-android-gradle-plugin#444]
### Describe the bug
We were using Bugsnag Gradle Plugin 7.0.0 + DexGuard Gradle Plugin 1.1.14 and we could build the project just fine. We're trying to update Bugsnag Gradle Plugin to 7.1.0 but it fails the build on the configuration phase.

We tried to update the DexGuard to the 1.1.15 (which was released today) and it failed with the same error.

This is the build output:

```
* What went wrong:
A problem occurred configuring project ':app'.
> Task with name 'dexguardAabColombiaRelease' not found in project ':app'.

...

at com.bugsnag.android.gradle.BugsnagPlugin.registerBugsnagTasksForVariant(BugsnagPlugin.kt:205)
	at com.bugsnag.android.gradle.BugsnagPlugin.access$registerBugsnagTasksForVariant(BugsnagPlugin.kt:67)
	at com.bugsnag.android.gradle.BugsnagPlugin$setupBugsnagPlugin$1$1.execute(BugsnagPlugin.kt:157)
```

### Steps to reproduce
You'll need a project with build variants + build flavors.

I'll try to build a sample and attach it here.

### Environment
* Gradle version: 7.2
* Android Gradle Plugin (AGP) version: 7.0.2
* Bugsnag Android Gradle Plugin version: 7.1.0
* DexGuard Gradle Plugin version: 1.1.14 and 1.1.15

**Comment by yousif-bugsnag** [comment]
Hi @guilherme-pereira-ifood, I've tried building a project with build variants and flavors, using those versions (Gradle 7.2, AGP 7.0.2, BAGP 7.1.0, DGP 1.1.15), but wasn't able to reproduce this. If you are able to provide a cut down reproduction case that demonstrates the issue that would be really useful.

**Comment by guilherme-pereira-ifood** [comment]
Hi @yousif-bugsnag.

I built a reproducible in this repository: https://github.com/guilherme-pereira-ifood/DexGuardAGP7Issue

This repository is using bugsnag gradle plugin 7.0.0. Just change it to 7.1.0 on root `build.gradle` and sync the project. You'll be able to see the error happening.

Please check `app/build.gradle` to understand what we may be doing differently from your tests.

**Comment by guilherme-pereira-ifood** [comment]
Hi @yousif-bugsnag.

I built a reproducible in this repository: guilherme-pereira-ifood/DexGuardAGP7Issue

This repository is using bugsnag gradle plugin 7.0.0. Just change it to 7.1.0 on root build.gradle and sync the project on AS. You'll be able to see the error happening.

Please check app/build.gradle to understand what we may be doing differently from your tests.

**Comment by mattdyoung** [comment]
@guilherme-pereira-ifood 

Thanks for the reproduction example. It looks like for a custom build type (that isn't debug/release) it's not creating the task eagerly. Prior to v7.1.0 Bugsnag would just skip uploading if it couldn't find the task, but it's now throwing an exception. We're looking into the best way to resolve this.

As a temporary workaround you could disable Bugsnag for the affected buildType. Note that in your example the `bugsnag` closure is in the wrong place, you'd need to move it outside the `android` closure and do something like this:
```
android {
    // android configuration here
}

bugsnag {
    variantFilter { variant ->
        def name = variant.name.toLowerCase()
        if (name.contains("debug") || name.contains("releasecandidate")) {
            enabled = false
        }
    }
}
```

**Comment by guilherme-pereira-ifood** [comment]
@mattdyoung I just tried with the suggested workaround, but the same issue happens.

```
Caused by: org.gradle.api.UnknownTaskException: Task with name 'dexguardAabColombiaRelease' not found in project ':app'. |  
-- | --
  | at org.gradle.api.internal.tasks.DefaultTaskCollection.createNotFoundException(DefaultTaskCollection.java:102) |  
  | at org.gradle.api.internal.tasks.DefaultTaskCollection.createNotFoundException(DefaultTaskCollection.java:46) |  
  | at org.gradle.api.internal.DefaultNamedDomainObjectCollection.named(DefaultNamedDomainObjectCollection.java:360) |  
  | at org.gradle.api.internal.tasks.DefaultTaskCollection.named(DefaultTaskCollection.java:112) |  
  | at com.bugsnag.android.gradle.BugsnagPlugin$registerBugsnagTasksForVariant$1.execute(BugsnagPlugin.kt:351) |  
  | at com.bugsnag.android.gradle.BugsnagPlugin$registerBugsnagTasksForVariant$1.execute(BugsnagPlugin.kt:67) |  
  | at org.gradle.configuration.internal.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:123) |  
  | at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction$1.run(DefaultCollectionCallbackActionDecorator.java:110) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68) |  
  | at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56) |  
  | at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:74) |  
  | at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45) |  
  | at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:74) |  
  | at org.gradle.api.internal.DefaultCollectionCallbackActionDecorator$BuildOperationEmittingAction.execute(DefaultCollectionCallbackActionDecorator.java:107) |  
  | at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:44) |  
  | at org.gradle.api.internal.DefaultMutationGuard$2.execute(DefaultMutationGuard.java:44) |  
  | at org.gradle.api.internal.DefaultDomainObjectCollection.configureEach(DefaultDomainObjectCollection.java:183) |  
  | at com.bugsnag.android.gradle.BugsnagPlugin.registerBugsnagTasksForVariant(BugsnagPlugin.kt:205) |  
  | at com.bugsnag.android.gradle.BugsnagPlugin.access$registerBugsnagTasksForVariant(BugsnagPlugin.kt:67) |  
  | at com.bugsnag.android.gradle.BugsnagPlugin$setupBugsnagPlugin$1$1.execute(BugsnagPlugin.kt:157)

```

**Comment by xander-jones** [comment]
Hey @guilherme-pereira-ifood 👋  `app:assembleBrazilRelease` doesn't contain `debug` or `releasecandidate` so this workaround won't work as Bugsnag's not being disabled, if you change the `build.gradle` setup as follows this should disable Bugsnag in the release that's failing above.

```gradle
android {
    // android configuration here
}

bugsnag {
    variantFilter { variant ->
        def name = variant.name.toLowerCase()
        if (name.contains("debug") || name.contains("releasecandidate") || name.contains("brazilrelease") {
            enabled = false
        }
    }
}
```

**Comment by guilherme-pereira-ifood** [comment]
I don't think that's a valid workaround since we'll disable bugsnag in production. Or am I missing something?

**Comment by johnkiely1** [comment]
Hey @guilherme-pereira-ifood, this would disable the Bugsnag Android Gradle Plugin so uploading of the mapping files would not happen automatically for your `brazilrelease` type. But Bugsnag would still report errors for your project. You could look to [upload the files manually](https://docs.bugsnag.com/api/android-mapping-upload/#uploading-mapping-files) for that release type so as to get the mapped traces in the dashboard.  

Having said that if v7.0.0 was working as expected you could revert to that until we can address this issue properly.

**Comment by luke-belton** [comment]
Hi @guilherme-pereira-ifood - this work is in our backlog however with current priorities we're not expecting to get to this immediately. We'll update this thread as soon as any changes are made. In the meantime we'd recommend using the workaround suggested above or reverting to `v7.0.0`.

**Comment by yousif-bugsnag** [comment]
This was is fixed in [v7.2.0](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v7.2.0)
